File: python/google_code_jam/2014/qualifications/cookies/cookies.cpp
#include <iostream>

using namespace std;


double solve(double C, double F, double X) {
    double r = 2.0;
    double t = 0.0;
    while(true) {
        if(C + (r * X) / (r + F) < X) {
            t += C / r;
            r += F;
        }
        else {
            return t + X / r;
        }
    }
}

int main() {
    int T;
    cin >> T;
    for(int t = 1; t <= T; t++) {
        double C, F, X;
        cin >> C >> F >> X;
        cout << "Case #" << t << ": " << solve(C, F, X) << endl;
    }
}

File: python/google_code_jam/2014/qualifications/cookies/cookies.py
from __future__ import print_function
from __future__ import division

import sys


__author__ = 'Sahand Saba'


def read_case(reader):
    C, F, X = (float(x) for x in next(reader).split())
    return C, F, X


def solve(C, F, X):
    r = 2.0
    t = 0.0
    while True:
        if C + (r * X) / (r + F) < X:
            t += C / r
            r += F
        else:
            return t + X / r


if __name__ == '__main__':
    reader = iter(sys.stdin)
    T = int(next(reader))
    for t in xrange(T):
        C, F, X = read_case(reader)
        s = solve(C, F, X)
        print('Case #{0}: {1} '.format(t + 1, s))
